### DISCUSSION TITLE
Can't boot from Infiniband SRP.

### DISCUSSION THREAD

**Discussion by gobiursus** [ipxe/ipxe#1226] [General]
Hi all,
First thing, thank you @mcb30 for your help on #1220 .
I successfully `sanhook` and add initiator on my SAN. But when I execute `sanboot` it just stuck there forever. At the meanwhile my SAN is not showing any I/O at all, there's also no log related with the issue in `dmesg -T` on my SAN.

This is what happened after I successfully did `sanhook` and execute `sanboot`, it stuck here forever:
![image](https://github.com/ipxe/ipxe/assets/48418918/a99f8023-d4e7-44f3-9688-1596b837c42d)

I compiled my iPXE with option `make  bin-x86_64-pcbios/15b31003.mrom DEBUG=scsi:3,iscsi,infiniband`.
My NIC(): MCX354-FCBT
My iPXE version: I cloned from Github on commit `dc118c5`

**Comment by NiKiZe** [comment]
In cases like this, using tcpdump might be helpful,
Maybe on purpose try to use incorrect Uri to make sure  you know the different types of error cases.

**Reply by gobiursus** [reply]
`tcpdump` doesn't support Infiniband, SRP is Scsi RDMA Protocol, it is impossible to capture RDMA packets because they skip CPU.
If Uri is incorrect it I won't be able to `sanhook`.

**Comment by gobiursus** [comment]
Can you help me to take a look?
@mcb30

**Comment by mcb30** [comment]
SRP uses Infiniband RC packets, rather than the UD packets used for IPoIB.  The `hermon` driver in iPXE was originally written for and tested with ConnectX-2 HCAs, and then extended to support ConnectX-3.  I don't know if anyone would have ever tested the RC functionality on ConnectX-3, because it's a pretty obscure use case.

Fixing this would require some relatively difficult debugging.  If you're happy to try working on this yourself, I would suggest starting by building with `DEBUG=hermon:3,ib_cm:3,ib_cmrc:3,ib_smc,infiniband`.

**Reply by gobiursus** [reply]
I notice the initiator GUID and max bytes that my SAN device received from iPXE are different with the same set up on Linux.

**Reply by gobiursus** [reply]
What's the feature difference between `hermon` and `golan`? Can we combine them together to make sure ConnectX3 can access to same feature of newer cards that was supported in `golan`?

**Comment by gobiursus** [comment]
If I use it as extra storage device on Linux, they will be working fine.

**Reply by mcb30** [reply]
> If I use it as extra storage device on Linux, they will be working fine.

That is irrelevant, since the driver code is unrelated to the Linux driver.
